### PR TITLE
Replacing jackson-databind dependency with Google Gson for JSON serialization

### PR DESCRIPTION
**Pull request by fabiobrz**
Here we're just replacing dependencies that RestAssured needs in order to serialize POJOs, see https://github.com/rest-assured/rest-assured/wiki/Usage#content-type-based-serialization.

Here's a link to a successful job run, although it's just for sanity check as Maven `mp-open-api` profile is not run by our TS test job now, so that was verified locally: https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-simple-face/7/

Please make sure your PR meets the following requirements:
- [x] Pull Request contains description for the changes
- [x] Pull Request does not include fixes for multiple issues / topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to passing job is provided

### CODE REVIEW

**Comment by fabiobrz** [comment]
Gson now to 2.8.5 after first Travis build failed several times on jdk11 with 2.8.6.

File: pom.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
         <version.org.wildfly.core>10.0.3.Final</version.org.wildfly.core>
         <version.org.wildfly.extras.creaper>1.6.1</version.org.wildfly.extras.creaper>
         <version.org.yaml.snakeyaml>1.24</version.org.yaml.snakeyaml>
-        <version.com.fasterxml.jackson.core.jackson-databind>2.9.10.1</version.com.fasterxml.jackson.core.jackson-databind>
+        <version.com.google.code.gson>2.8.5</version.com.google.code.gson>
```
Comment (mnovak1) [inline pom.xml]: There is `2.8.6` already, could you update it, please?

Comment (fabiobrz) [reply pom.xml]: 2.8.6 failed several (2 or 3) attempts for Travis checks yesterday on jdk11. Then I bumped it to 2.8.5: https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/88#issuecomment-564674003

Comment (mnovak1) [reply pom.xml]: Looks like it was cause by network issue in travis when downloading it. Version was updated and PR merged.

**Pull request merged**